### PR TITLE
Rework x11_start_program to retry mistyping

### PR DESCRIPTION
**Pull request by DrMullings**
Now retrying up to three times if command was not found

- Related ticket: https://progress.opensuse.org/issues/55580

### CODE REVIEW

**Comment by DrMullings** [comment]
Verification run:
https://openqa.suse.de/tests/3346981#step/firefox/4

**Comment by DrMullings** [comment]
@SergioAtSUSE please take a look

File: lib/susedistribution.pm [hunk 1]
```diff
@@ -256,34 +256,40 @@ sub x11_start_program {
...
+            assert_screen([@target], $args{match_timeout}, no_wait => $args{match_no_wait});
+            last unless match_has_tag('desktop-runner-border') || match_has_tag('desktop-runner-plasma-suggestions');
+            wait_screen_change {
+                send_key 'ret';
+            };
+        }
+        set_var('IN_X11_START_PROGRAM', undef);
+        # asserting program came up properly
+        last if match_has_tag @target;
+        next if match_has_tag('desktop-runner-border') || match_has_tag('desktop-runner-plasma-suggestions');
```
Comment (foursixnine) [inline lib/susedistribution.pm]: You could combine both into the same call? (I know it's old code, but a little cleanup never hurt anybody :D)
```suggestion
        next if match_has_tag qw(desktop-runner-border desktop-runner-plasma-suggestions);
```

**Comment by foursixnine** [comment]
> Verification run:
> https://openqa.suse.de/tests/3346981#step/firefox/4

Give it a look at [this pr](https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/8192/files#r313780920) and the comments (resolved), since just verifying that the good case works is not going to cut it, as the alternate code path should be hit too... 

Long story short: alter the schedule from an extratest scenario boot from the hdd, and trigger the firefox test or firefox audio test about 10/100 times within the same run? :) aarch64 should quickly hit this...

**Comment by DrMullings** [comment]
https://openqa.suse.de/tests/3363723
Here we go

**Comment by DrMullings** [comment]
https://openqa.suse.de/tests/3363758#settings
and a second run to double the chances

File: lib/susedistribution.pm (hunk 1 above)
Comment (foursixnine) [reply lib/susedistribution.pm]: Thanks for taking this into account!, but you also missed the same thing, couple of lines above... https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/8403/files#diff-a2a433d5914343b617ae52ac0fb40524R282

**Comment by foursixnine** [comment]
You need to tidy up :)

File: lib/susedistribution.pm [hunk 2]
```diff
@@ -256,34 +256,40 @@ sub x11_start_program {
...
+            assert_screen([@target], $args{match_timeout}, no_wait => $args{match_no_wait});
+            last unless match_has_tag('desktop-runner-border') || match_has_tag('desktop-runner-plasma-suggestions');
+            wait_screen_change {
+                send_key 'ret';
+            };
+        }
+        set_var('IN_X11_START_PROGRAM', undef);
+        # asserting program came up properly
+        last                                                                if match_has_tag @target;
+        next                                                                if match_has_tag qw(desktop-runner-border desktop-runner-plasma-suggestions);
```
Comment (foursixnine) [inline lib/susedistribution.pm]: @DrMullings Could you tidy up again with this changes? to see how the code looks like? Tidy rules need to be changed... seriously :(

```suggestion
my @runner_tags = qw(desktop-runner-border desktop-runner-plasma-suggestions);
next if match_has_tag @runner_tags;
```

File: lib/susedistribution.pm [hunk 3]
```diff
@@ -256,34 +256,40 @@ sub x11_start_program {
...
+        # As above especially krunner seems to take some time before disappearing
+        # after 'ret' press we should wait in this case nevertheless
+        wait_still_screen(3, similarity_level => 45) unless ($args{no_wait} || ($args{valid} && $args{target_match} && !check_var('DESKTOP', 'kde')));
+        return unless $args{valid};
+        set_var('IN_X11_START_PROGRAM', $program);
+        my @target = ref $args{target_match} eq 'ARRAY' ? @{$args{target_match}} : $args{target_match};
+        for (1 .. 3) {
+            push @target, check_var('DESKTOP', 'kde') ? 'desktop-runner-plasma-suggestions' : 'desktop-runner-border';
+            assert_screen([@target], $args{match_timeout}, no_wait => $args{match_no_wait});
+            last unless match_has_tag('desktop-runner-border') || match_has_tag('desktop-runner-plasma-suggestions');
```
Comment (foursixnine) [inline lib/susedistribution.pm]: could you apply here the same solution as L290?

**Comment by DrMullings** [comment]
I can't tidy from home, but the code should be adapted

**Comment by DrMullings** [comment]
Again ready to be reviewed

File: lib/susedistribution.pm (hunk 3 above)
Comment (foursixnine) [reply lib/susedistribution.pm]: @DrMullings wanna take this one too? (I won't block merging if you don't want to :P, but would be nice :D )

Comment (DrMullings) [reply lib/susedistribution.pm]: Sorry, I completely overlooked this, I guess :D

Comment (DrMullings) [reply lib/susedistribution.pm]: Now it should be there

**Comment by DrMullings** [comment]
Can be reviewd again ;)

**Comment by foursixnine** [comment]
For the record, this can be merged :),

**Pull request merged**